### PR TITLE
Arc - support static producer methods.

### LINKED ISSUE #5513: ArC - support static producers
See https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#producer_method.

### PR DESCRIPTION
**Pull request by manovotn**
Fixes #5513

### CODE REVIEW

File: independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java [hunk 1]
```diff
@@ -1173,8 +1173,15 @@ protected void implementCreate(ClassOutput classOutput, ClassCreator beanCreator
                         paramTypesArray,
                         declaringProviderInstanceHandle, argsArray));
             } else {
-                create.assign(instanceHandle, create.invokeVirtualMethod(MethodDescriptor.of(producerMethod),
-                        declaringProviderInstanceHandle, referenceHandles));
+                ResultHandle invokeMethodHandle;
+                if (Modifier.isStatic(producerMethod.flags())) {
```
Comment (mkouba) [inline independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java]: For static producers we don't have to allocate the `declaringProviderInstanceHandle` at all. And we need to fix the reflection fallback for private methods too.

**Review by mkouba — CHANGES_REQUESTED** [review]
The issue mentions "static producers" which includes static producer fields ;-).

**Review by manovotn — COMMENTED** [review]
@mkouba how about this version?
Sorry for missing the fields initially, I was just coding without thinking :)

**Comment by manovotn** [comment]
The CI failure looks very much unrelated.

**Review by mkouba — APPROVED** [review]

**Pull request merged**